worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -30,28 +58,159 @@ public Job putJobInProgress(String jobId) {
         final Job job = jobRepository.findByJobUuid(jobId);
         Assert.notNull(job, String.format("Job %s not found", jobId));
 
-
         // validate status is SUBMITTED
         if (!SUBMITTED.equals(job.getStatus())) {
-            final String errMsg = String.format("Job %s is not in %s status. Skipping job", jobId, SUBMITTED);
+            final String errMsg = String.format("Job %s is not in %s status.", jobId, SUBMITTED);
             throw new IllegalArgumentException(errMsg);
         }
 
         job.setStatus(IN_PROGRESS);
 
-        log.info("Job [{}] is now IN_PROGRESS", job.getId());
         return jobRepository.save(job);
     }
 
     @Override
-    public void completeJob(Job job) {
+    @Transactional(propagation = Propagation.NEVER)
+    public Job processJob(final String jobId) {
+
+        final Job job = jobRepository.findByJobUuid(jobId);
+        log.info("Found job : {} - {}", job.getId(), job.getJobUuid());
+
+        final Sponsor sponsor = job.getUser().getSponsor();
+
+        final List<Contract> attestedContracts = getAggregatedAttestedContracts(sponsor);
+        log.info("Job [{}] has [{}] attested contracts", job.getJobUuid(), attestedContracts.size());
+
+
+        final Path outputDirPath = Paths.get(efsMount, job.getJobUuid());
+        final Path outputDir = fileService.createDirectory(outputDirPath);
+
+        for (Contract contract : attestedContracts) {
+            log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
+
+            var jobOutputs = processContract(outputDir, contract);
+
+            jobOutputs.forEach(jobOutput -> job.addJobOutput(jobOutput));
+            jobOutputRepository.saveAll(jobOutputs);
+        }
```

Comment:
So any uncaught exception here will terminate the job processing (which is what we need), but it won't set the job's status to `FAILED`. Let's add a try/catch block that does so.

For `FAILED` jobs we don't need to create JobOutput records. Let's write the exception message to `Job.statusMessage` though.

Reply:
done